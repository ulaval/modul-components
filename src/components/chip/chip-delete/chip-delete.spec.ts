import { mount, Wrapper } from '@vue/test-utils';
import Vue from 'vue';
import { renderComponent } from '../../../../tests/helpers/render';
import uuid from '../../../utils/uuid/uuid';
import MChipDeletePlugin, { MChipDelete } from './chip-delete';

jest.mock('../../../utils/uuid/uuid');
(uuid.generate as jest.Mock).mockReturnValue('uuid');

describe('Chip', () => {
    beforeEach(() => {
        Vue.use(MChipDeletePlugin);
    });
    it('should render correctly', () => {
        const component: Wrapper<MChipDelete> = mount(MChipDelete, {
            localVue: Vue
        });

        return expect(renderComponent(component.vm)).resolves.toMatchSnapshot();
    });
});
