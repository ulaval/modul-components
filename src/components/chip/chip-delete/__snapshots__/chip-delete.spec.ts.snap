// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render correctly 1`] = `
<div class="m-chip-delete">
  <div class="m-chip-delete__body"><span id="mChipDeleteText-uuid" class="m-chip-delete__text"><span class="m-chip-delete__hidden-text">chip:deleteÂ : </span> </span> <button aria-labelledby="mChipDeleteText-uuid" class="m-chip-delete__button">
      <m-icon size="14px" name="m-svg__close-clear" class="m-chip-delete__icon"></m-icon>
    </button></div>
</div>
`;
